### PR TITLE
[#156877577] Make the password field optional

### PR DESCRIPTION
**Pull request by iankigen**
#### What does this PR do?

- Make the password field optional 

#### Description of Task to be completed?

- add user creation form

#### How should this be manually tested?

`python manage.py test`

#### What are the relevant pivotal tracker stories?

[ #156877577](https://www.pivotaltracker.com/n/projects/2146417/stories/156877577)

#### Screenshots
no_pass
no_pass_admin

### CODE REVIEW

**Comment by egichuri** [comment]
@iankigen i think the password field is required by default. so we might need to override that on the `User` model as well

**Comment by iankigen** [comment]
@egichuri by removing `password` from `UserSerializer` in my previous task, we could create a User without a password through the APi.

**Comment by egichuri** [comment]
@iankigen yea.. but we need the db to allow that as well for the sake of users created outside of the API and admin form

**Comment by HawiCaesar** [comment]
@egichuri in what case will a user be created outside the API and admin form ?

**Comment by egichuri** [comment]
eg fixtures

File: api/models.py [hunk 1]
```diff
@@ -168,6 +168,7 @@ class User(AbstractUser):
     picture = models.CharField(max_length=255, blank=True, null=True)
     phone_number = models.CharField(max_length=50, blank=True, null=True)
     last_modified = models.DateTimeField(auto_now=True, editable=False)
+    password = models.CharField(max_length=128, blank=True, default='')
```
Comment (egichuri) [inline api/models.py]: we could do `null=True` and get rid of the default

Comment (iankigen) [reply api/models.py]: on it

**Comment by iankigen** [comment]
@egichuri done

**Pull request merged**